### ISSUE TITLE
`TestMethodValidatorTests` are not testing what they are supposed to test.

### ISSUE THREAD

**Issue by Youssef1313** [microsoft/testfx#4207]
*All* tests in there that use mocked method info are not setting up IsPublic to return true, meaning that ISValidTestMethod returns false just because IsPublic is false regardless of the scenario meant by other setup parts.

For example (putting aside that we are working on supporting generics), the tests that are asserting generic methods return false will still pass if that part regressed, because it will then return false still as IsPublic is false.

**Comment by nohwnd** [comment]
Are those some of the mocked up tests? I had the same feeling, and started doing this, but failed to finish for lack of time. And also because we test a lot of things that would clash without assembly isolation. https://github.com/microsoft/testfx/pull/2925

**Comment by Youssef1313** [comment]
@nohwnd Yup, they are the mocked tests. The PR you referenced touched the ones I was referring to (and more). It would be nice to fix these up.
